src/main/java/backend/Logic.java

```diff
@@ -331,6 +331,30 @@ private void revertLocalLabelsReplace(TurboIssue modifiedIssue, List<String> ori
     }
 
     /**
+     * Edit the state of an issue on GitHub
+     *
+     * @param issue The issue whose state is to be updated
+     * @param open The new state for the issue
+     * @return True for success, false otherwise
+     */
+    public CompletableFuture<Boolean> editIssueState(TurboIssue issue, boolean open) {
+        String action = open ? "Reopening" : "Closing";
+        logger.info(String.format("%s %s on GitHub", action, issue));
+        return repoIO.editIssueState(issue, open)
+                .thenApply(success -> {
+                    if (success) {
+                        logger.info("Changing state for " + issue + " on UI");
+                        issue.setOpen(open);
```

Comment:
It's unsafe to modify a dangly reference to `TurboIssue` to change the state of repositories. You can refer to `replaceIssueLables` method to see how we currently modify a local state. You can also take a look at the header comment of `MultiModel`